migration/mule/foreach-processing-and-choice-routing/foreachProcessingandchoicerouting/src/main/synapse-config/api/Loan.xml

```diff
@@ -0,0 +1,110 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<api context="/loan" name="Loan" xmlns="http://ws.apache.org/ns/synapse">
+    <resource methods="GET" uri-template="/calculate*">
+        <inSequence>
+            <property expression="boolean(get-property('query.param.name')) and boolean(get-property('query.param.amount')) and boolean(get-property('query.param.term')) and boolean(get-property('query.param.ssn')) " name="containAllQueryParams" scope="default" type="STRING"/>
+            <switch source="$ctx:containAllQueryParams">
+                <case regex="true">
+                    <call>
+                        <endpoint key="HttpEndpoint"/>
+                    </call>
+                    <foreach expression="json-eval($.banks)" id="ForEachBank">
+                        <sequence>
+                            <switch source="$body/jsonValue">
```

Comment:
Could you please explain what we do by this expression `<switch source="$body/jsonValue">`? I could not find where it gets set.

Reply:
Explained in the descriptions of the mediators.